mm2src/coins/eth/v2_activation.rs

```diff
@@ -263,3 +267,29 @@ pub async fn eth_coin_from_conf_and_request_v2(
 
     Ok(EthCoin(Arc::new(coin)))
 }
+
+/// Processes the given `priv_key_policy` and generates corresponding `KeyPair`.
+/// This function expects either [`PrivKeyBuildPolicy::IguanaPrivKey`]
+/// or [`PrivKeyBuildPolicy::GlobalHDAccount`], otherwise returns `PrivKeyPolicyNotAllowed` error.
+pub(crate) fn key_pair_from_priv_key_policy(
+    conf: &Json,
+    priv_key_policy: PrivKeyBuildPolicy,
+) -> MmResult<KeyPair, EthActivationV2Error> {
+    let priv_key = match priv_key_policy {
```

Comment:
I agree, it could be easier to implement it.
The reason why I decided to add new `PrivKeyBuildPolicy` is to not parse the coins config in methods that shouldn't do it, for example, `coins_activation::platform_coin_with_tokens`.
I'll try this approach

Reply:
I've tried to remove `PrivKeyBuildPolicy::GlobalHDAccount` at `mm2-hd-account-secp-secret` branch, the diff is:
https://github.com/KomodoPlatform/atomicDEX-API/compare/mm2-hd-account...KomodoPlatform:atomicDEX-API:mm2-hd-account-secp-secret

It didn't help to make fewer changes. Especially functions like `eth_coin_from_conf_and_request` needs to take `PrivKeyBuildPolicy` due to the fact that we plan to integrate `MetaMask`, and I think it will be done through the new `PrivKeyBuildPolicy::MetaMask` enum variant.

Please compare `mm2-hd-account` and `mm2-hd-account-secp-secret` branches and tell which approach is better.
I'm ok with both

Reply:
> It didn't help to make fewer changes.
@sergeyboyko0791 Comparing with dev https://github.com/KomodoPlatform/atomicDEX-API/compare/dev...KomodoPlatform:atomicDEX-API:mm2-hd-account-secp-secret, there are 100 fewer additions and deletions 🙂 

> and I think it will be done through the new PrivKeyBuildPolicy::MetaMask enum variant

Does it mean that we will have to handle this Metamask variant, even for coins that might not support it all?

Reply:
I supposed that MetaMask Snaps don't allow to reveal the private key outside the sandbox environment, but it turned out that it's possible.
Here is the proof: https://github.com/KomodoPlatform/atomicDEX-API/issues/1167#issuecomment-1309328693
> Does it mean that we will have to handle this Metamask variant, even for coins that might not support it all?

So before I planned that our custom aDEX Snap will sign transactions, return public keys, so coins like UTXO will communicate with it through `PrivKeyBuildPolicy::MetaMask`.
But since it's (currently) possible to just get a private key, we don't actually need to add `PrivKeyBuildPolicy::MetaMask` for all coins.

However we still need to pass `PrivKeyBuildPolicy::MetaMask` to ETH/ERC20 coins.
Do you think we can implement a workaround, so the `MetaMask` policy can be used for ETH/ERC20 coins only?
Probably, we can add `EthPrivKeyBuildPolicy` enum 🤔

Reply:
Merged into `mm2-hd-account` branch
> Comparing with dev https://github.com/KomodoPlatform/atomicDEX-API/compare/dev...KomodoPlatform:atomicDEX-API:mm2-hd-account-secp-secret, there are 100 fewer additions and deletions 🙂

Reply:
> However we still need to pass PrivKeyBuildPolicy::MetaMask to ETH/ERC20 coins.
Do you think we can implement a workaround, so the MetaMask policy can be used for ETH/ERC20 coins only?
Probably, we can add EthPrivKeyBuildPolicy enum 🤔

I think it's worth trying to add `EthPrivKeyBuildPolicy` for the Metamask integration 🙂